server/tserver/src/main/java/org/apache/accumulo/tserver/session/SessionManager.java

```diff
@@ -458,34 +458,37 @@ public List<ActiveScan> getActiveScans() {
         addActiveScan(activeScans, scanSession,
             isSingle ? ((SingleScanSession) scanSession).extent
                 : ((MultiScanSession) scanSession).threadPoolExtent,
-            ct, isSingle ? ScanType.SINGLE : ScanType.BATCH,
-            computeScanState(scanSession.getScanTask()), scanSession.scanParams, entry.getKey());
+            ct, isSingle ? ScanType.SINGLE : ScanType.BATCH, computeScanState(scanSession),
+            scanSession.scanParams, entry.getKey());
       }
     }));
 
     return activeScans;
   }
 
-  private ScanState computeScanState(ScanTask<?> scanTask) {
+  private ScanState computeScanState(ScanSession<?> session) {
     ScanState state = ScanState.RUNNING;
 
-    if (scanTask == null) {
-      state = ScanState.IDLE;
+    if (session.getState() == State.REMOVED) {
+      state = ScanState.CLEANING;
     } else {
-      switch (scanTask.getScanRunState()) {
+      switch (session.getScanTask().getScanRunState()) {
         case QUEUED:
           state = ScanState.QUEUED;
           break;
         case FINISHED:
           state = ScanState.IDLE;
           break;
         case RUNNING:
+          if (session.getState() == State.REMOVED) {
+            state = ScanState.ZOMBIE;
+          }
+          break;
```

Comment:
The current logic is:
```
    if (session.getState() == State.REMOVED) {
      state = ScanState.CLEANING;
    } else {
      ...
      if (session.getState() == State.REMOVED) {
        state = ScanState.ZOMBIE;
      }
      ...
    }
```
`state = ScanState.ZOMBIE` will never execute in this logic. It seems like either (or both) the logic for identifying ZOMBIE or CLEANING need to be changed (but I'm not sure which or what it should be changed to)